src/web3/ContractKitSigner.ts

```diff
@@ -0,0 +1,51 @@
+import { RLPEncodedTx } from '@celo/connect'
+import { EIP712TypedData } from '@celo/utils/lib/sign-typed-data-utils'
+import { LocalSigner } from '@celo/wallet-local'
+import BigNumber from 'bignumber.js'
+import Logger from 'src/utils/Logger'
+import { KeychainAccount } from 'src/web3/types'
+
+const TAG = 'web3/ContractKitSigner'
+
+/**
+ * Implements the signer interface on top of the OS keychain
+ */
+export default class ContractKitSigner extends LocalSigner {
```

Comment:
I think that would be up to later developers who flesh out the signing methods when they need to be used. Its possible that it could make the code cleaner when doing the switch over.